pandas/io/parsers.py

```diff
@@ -407,11 +408,15 @@ def _validate_names(names):
     Raises
     ------
     ValueError
-        If names are not unique.
+        If names are not unique or have incosistent ordering (e.g. set).
     """
     if names is not None:
         if len(names) != len(set(names)):
             raise ValueError("Duplicate names are not allowed.")
+        if not is_list_like(names, allow_sets=False):
+            raise ValueError(
+                "Names should have consistent ordering. Consider a list instead."
```

Comment:
Wording nit, but something like "names should be an ordered collection" sounds a bit better / more correct IMHO (I wouldn't describe actually having an order as having a "consistent" order)

Reply:
Thanks for your comment! The fact that there is something called "OrderedSet" suggests that the word "ordered" alone can mean the same thing that I meant by "consistent" but It's more Pythonic. I'll update the message